torch/_functorch/aot_autograd.py

```diff
@@ -1463,7 +1463,14 @@ def view_avoid_dupes_with_primals(t):
         f_input_tangents = [
             inp
             for inp, info in zip(flat_f_args, input_info)
-            if info.mutates_data and info.requires_grad
+            if _get_mutation_type(
+                keep_input_mutations,
+                mutates_data=info.mutates_data,
+                mutates_metadata=info.mutates_metadata,
+                requires_grad=info.requires_grad
+                # MUTATED_OUT_GRAPH corresponds to any input mutations that happen outside the graph.
+                # this can also include metadata mutations, and inputs that do not require grad,
+            ) == MutationType.MUTATED_OUT_GRAPH and info.mutates_data and info.requires_grad
```

Comment:
What would happen for inputs that are marked with MutationType.MUTATED_OUT_GRAPH but not fall under "info.mutates_data and info.requires_grad"

Reply:
An example would be an input that had a metadata mutation - we always keep metadata mutations outside of the graph, regardless of whether they require grad.

In that case, we still don't want to create a tangent (backward input) for that input (we only need to create tangents for inputs that require grad and have data mutations).

I think the main reason I added this extra check is because @oulgen is doing some work to add mutations that can be "hidden" from autograd, that we want to keep in the graph. And for that case, we also don't want to create tangents for those updated inputs in the backward.